components/local-app/pkg/bastion/bastion.go

```diff
@@ -597,5 +622,90 @@ func (b *Bastion) doTunnelPorts(ws *Workspace) error {
 			}
 		}
 		ws.tunnelListenersMu.Unlock()
+		b.notify(ws)
+	}
+}
+
+func (b *Bastion) notify(ws *Workspace) {
+	b.subscriptionsMu.RLock()
+	defer b.subscriptionsMu.RUnlock()
+	var subs []*Subscription
+	for sub := range b.subscriptions {
+		if sub.instanceID == ws.InstanceID {
+			subs = append(subs, sub)
+		}
 	}
+	if len(subs) <= 0 {
+		return
+	}
+	status := ws.Status()
+	for _, sub := range subs {
+		select {
+		case sub.updates <- status:
+		case <-time.After(5 * time.Second):
+			log.Error("ports subscription droped out")
+			sub.Close()
+		}
+	}
+}
+
+func (b *Bastion) Status(instanceID string) []*app.TunnelStatus {
+	ws, ok := b.getWorkspace(instanceID)
+	if !ok {
+		return nil
+	}
+	return ws.Status()
+}
+
+func (ws *Workspace) Status() []*app.TunnelStatus {
```

Comment:
/nit maybe move out of the Bastion methods?

Reply:
moved closed to Workspace